src/components/flash/flash.js

```diff
@@ -456,34 +465,38 @@ class Flash extends React.Component {
    * @return {Object} JSX
    */
   render() {
-    let flashHTML, sliderHTML;
-
-    if (this.props.open) {
-      flashHTML = this.flashHTML;
-      sliderHTML = this.sliderHTML;
-    }
-
+    const sliderHTML = this.props.open && this.sliderHTML;
+    const flashHTML = this.props.open && this.flashHTML;
     return (
-      <div { ...tagComponent('flash', this.props) }>
-        <div className={ this.classes }>
-          <CSSTransitionGroup
-            transitionName='carbon-flash__slider'
-            transitionEnterTimeout={ 600 }
-            transitionLeaveTimeout={ 600 }
-          >
-            { sliderHTML }
-            <CSSTransitionGroup
-              transitionName='carbon-flash__content'
-              transitionEnterTimeout={ 800 }
-              transitionLeaveTimeout={ 500 }
-            >
-              { flashHTML }
-            </CSSTransitionGroup>
-          </CSSTransitionGroup>
-        </div>
-
-        { this.dialogs }
-      </div>
+      this.state.open && (
+        <Portal>
+          <div { ...tagComponent('flash', this.props) }>
+            <div className={ this.classes }>
+              <CSSTransitionGroup
+                transitionAppear
+                transitionAppearTimeout={ 500 }
+                transitionName='carbon-flash__slider'
+                transitionEnterTimeout={ 600 }
+                transitionLeave
+                transitionLeaveTimeout={ 2600 }
+              >
+                { sliderHTML }
+                <CSSTransitionGroup
+                  transitionAppear
```

Comment:
And again: you'll need a `component='div'` prop here, otherwise the HTML will be invalid.

See #1708.